File: C#_Base_PT4/XFile52.cs
// File: "XFile52"
using PT4;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.IO;

namespace PT4Tasks
{
    public class MyTask : PT
    {
        public static void Solve()
        {
            Task("XFile52");

            string s0 = GetString();

            FileStream g = new FileStream(s0, FileMode.Create);

            int N = GetInt();

            FileStream[] f = new FileStream[N];
            for(int k= 0; k < N; k ++){
                f[k] = new FileStream(GetString(), FileMode.Open);
            }

            BinaryReader[] r = new BinaryReader[N];
            for (int i = 0; i < N; i++)
                r[i] = new BinaryReader(f[i]);

            BinaryWriter w = new BinaryWriter(g);


            w.Write(N);
            for(int k = 0; k < N; k ++){
                int x = (int)f[k].Length/sizeof(int);
                w.Write(x);
            }

            for(int k = 0; k < N; k ++)
                while (f[k].Position < f[k].Length)
                    w.Write(r[k].ReadInt32());

            for (int i = 0; i < N; i++)
                r[i].Close();
            w.Close();
        }
    }
}

File: C++_Base_PT4/XFile52.cpp
#include "pt4.h"
#include <fstream>
using namespace std;

void Solve()
{
    Task("XFile52");
    
    string s0;
    
    pt >> s0;
    
    fstream g;
	g.open(s0, ios::binary | ios::out);
    
    int N;
    
    pt >> N;
	
	fstream *f = new fstream[N];
	
	for (int i = 0; i < N; ++i)
	{
		string s;
 		pt >> s;

 		f[i].open(s, ios::binary | ios::in);
	}
	
	g.write((char *)&N, sizeof(N));
	
	for(int i = 0; i < N; ++i)
	{
		f[i].seekg(0, ios::end);
		int len = f[i].tellg();
		int k = len/sizeof(int);
		g.write((char *)&k, sizeof(k));
		f[i].seekg(0, ios::beg);
	}

	for (int i = 0; i < N; ++i)
	{
		while (f[i].peek() != -1)
 		{
 			int x;
 			f[i].read((char *)&x, sizeof(x));
 			g.write((char *)&x, sizeof(x));
		}
 	}

	g.close();
	for (int i = 0; i < N; ++i)
 		f[i].close();
	delete[] f;
}
